### PR TITLE
test:fix broken unit test and add more test cases

### PR DESCRIPTION
**Pull request by chloeyin**
1. Fix broken unit tests except plugin_test.go, provider_test.go and files under verification package
2. Test all signing algorithms for signer test and verifier test
3. add algorithm test and envelope test

Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

### CODE REVIEW

File: signature/plugin.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func NewSignerPlugin(runner plugin.Runner, keyID string, pluginConfig map[string
 	return signer, nil
 }
 
-// Sign signs the artifact described by its descriptor, and returns the signature.
+// Sign signs the artifact described by its descriptor, and returns the marshalked envelope
```
Comment (patrickzheng200) [inline signature/plugin.go:52]: a typo

File: signature/signer_test.go [hunk 2]
```diff
@@ -31,73 +73,81 @@ func TestSignWithCertChain(t *testing.T) {
 	for _, k := range []int{2048, 3072, 4096} {
 		pk, _ := rsa.GenerateKey(rand.Reader, k)
 		certTuple := testhelper.GetRSACertTupleWithPK(pk, "TestSignWithCertChain_"+strconv.Itoa(pk.Size()), &rsaRoot)
```
Comment (patrickzheng200) [inline signature/signer_test.go:75]: testhelper now supports [GetRSACertTuple](https://github.com/notaryproject/notation-core-go/blob/9757c1e362333bbc6186095edc171c74914f428a/testhelper/certificatetest.go#L213) with different key sizes.

File: signature/signer_test.go [hunk 3]
```diff
@@ -31,73 +73,81 @@ func TestSignWithCertChain(t *testing.T) {
...
+			t.Run(fmt.Sprintf("RSA certificates of size %d", pk.Size()), func(t *testing.T) {
+				validateSignWithCerts(t, envelopeType, pk, []*x509.Certificate{certTuple.Cert, rsaRoot.Cert})
+			})
+		}
 	}
 
 	ecRoot := testhelper.GetECRootCertificate()
 	for _, v := range []elliptic.Curve{elliptic.P256(), elliptic.P384(), elliptic.P521()} {
 		pk, _ := ecdsa.GenerateKey(v, rand.Reader)
 		certTuple := testhelper.GetECDSACertTupleWithPK(pk, "TestSignWithCertChain_"+strconv.Itoa(pk.Params().BitSize), &ecRoot)
```
Comment (patrickzheng200) [inline signature/signer_test.go:86]: same here: [GetECCertTuple](https://github.com/notaryproject/notation-core-go/blob/9757c1e362333bbc6186095edc171c74914f428a/testhelper/certificatetest.go#L225)

**Review by patrickzheng200 — APPROVED** [review]
LGTM

File: signature/envelope_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	if expected == nil && got != nil {
+		return false
+	}
+	if expected != nil && got == nil {
+		return false
+	}
+	if expected == nil && got == nil {
+		return true
+	}
+	return expected.Error() == got.Error()
```
Comment (patrickzheng200) [inline signature/envelope_test.go:41]: how about:
```	
if expected == nil && got == nil {
	return true
}
if expected != nil && got != nil {
	return expected.Error() == got.Error()
}
return false
```

**Comment by binbin-li** [comment]
can you share the test coverage?

**Comment by chloeyin** [comment]
> can you share the test coverage?

This pr only includes a part of the tests to make code review easier. I will put coverage in later pr.

**Pull request merged**